test/services/blockchain/contracts/VestingContract.integration.test.ts

```diff
@@ -710,6 +712,133 @@ describe('VestingContract class', () => {
+            expect(balances.allocationsDetails[0].unlockAtVestingStart).toBe(0);
+        
+            const trx2 = await vestingContract.assignTokens('coinsale.tmy', accountName, '2.000000 LEOS', 999, signer);
+
+            expect(trx2.processed.receipt.status).toBe('executed');
+            balances = await vestingContract.getVestingAllocations(accountName);
+            expect(balances.allocationsDetails.length).toBe(2);
+            expect(balances.totalAllocation).toBe(4);
+
+        });
+
+    });
+
+    describe('vesting progress for unlockable coins getVestingAllocations()', () => {
+        test('Track vesting progress for category 998', async () => {
+            expect.assertions(5);
+            const { user } = await createRandomID();
+            const accountName = (await user.getAccountName()).toString();
+         
+            // Assign tokens to the account with vesting category 998
+            const trx = await vestingContract.assignTokens('coinsale.tmy', accountName, '4.000000 LEOS', 998, signer);
+        
+            expect(trx.processed.receipt.status).toBe('executed');
+        
+            // Fetch initial balances before vesting starts
+            let balances = await vestingContract.getVestingAllocations(accountName);
+        
+            // Ensure initial values are correct
+            expect(balances.totalAllocation).toBe(4);
+            expect(balances.unlockable).toBe(0);
+            expect(balances.unlocked).toBe(0);
+            expect(balances.locked).toBe(4);
+        
+            // Fetch vesting periods for category 998
+            const allocations = await vestingContract.getAllocations(accountName);
+            const vestingPeriod = VestingContract.calculateVestingPeriod(settings, allocations[0]);
+        
+            const { vestingStart, vestingEnd } = vestingPeriod;
+        
+            const totalAllocation = balances.totalAllocation;
+            const vestingProgress = [];
+
+            // Loop every second until vesting ends
+            for (let currentTime = vestingStart.getTime(); currentTime <= vestingEnd.getTime(); currentTime += 1000) {
+                await sleepUntil(new Date(currentTime));
+        
+                // Fetch updated balances
+                balances = await vestingContract.getVestingAllocations(accountName);
+        
+                const allocationDetails = balances.allocationsDetails[0];
+                const unlockablePercentage = (allocationDetails.unlockable / totalAllocation) * 100;
+                const unlockedPercentage = (allocationDetails.unlocked / totalAllocation) * 100;
+                const lockedPercentage = (allocationDetails.locked / totalAllocation) * 100;
+
+                // Store values in array
+                vestingProgress.push({
+                    time: new Date(currentTime).toISOString(),
+                    unlockable: `${allocationDetails.unlockable.toFixed(6)} LEOS (${unlockablePercentage.toFixed(2)}%)`,
```

Comment:
`${unlockablePercentage.toFixed(2)}`
just show the percentage number is fine. also for unlocked & locked